scripts/zones/Carpenters_Landing/mobs/Para.lua

```diff
@@ -0,0 +1,81 @@
+-----------------------------------
+-- Area: Carpenters Landing
+-- Mob:  Para
+-----------------------------------
+local ID = zones[xi.zone.CARPENTERS_LANDING]
+-----------------------------------
+
+local function spawnNextMob(player)
+    local mobId = GetNPCByID(ID.npc.qm_para):getLocalVar('nextMob')
+
+    if mobId ~= 0 then
+        GetMobByID(mobId):setSpawn(player:getXPos()-1, player:getYPos()-1, player:getZPos(), player:getRotPos() - 180) -- spawn facing by player
+        SpawnMob(mobId):updateEnmity(player)
+
+        if mobId < ID.mob.PARA_OFFSET + 4 then
+            GetNPCByID(ID.npc.qm_para):setLocalVar('nextMob', mobId + 1)
+        else
+            GetNPCByID(ID.npc.qm_para):setLocalVar('nextMob', 0)
+        end
+    end
+end
+
+local function AllParaDead()
+    for i = 0, 4 do
+        local mob = GetMobByID(ID.mob.PARA_OFFSET + i)
+        if not mob:isDead() then
+            return false
+        end
+    end
+
+    return true
+end
+
+local entity = {}
+
+entity.onMobInitialize = function(mob)
+    mob:setMobMod(xi.mobMod.IDLE_DESPAWN, 180) -- 3 min despawn
+    --mob:addImmunity(xi.immunity.LIGHT_SLEEP) immunity does not exist in status.lua
```

Comment:
There is no "Light sleep" effect. There is sleep effect from a light elemental source.

If the mob never gets slept by lullabies or repose, just set its light resistance rank to 11.

To expand:
An actual immunity would return the message `Para completely resists the effect of Sleep.`

The effect not landing, always returning `Para resists the effect of sleep` means it isnt actually immune, it just wont ever land, becouse the resistance rank makes the macc vs meva check always fail
- Resistance rank 10 floors magic hit rate to 5%
- Resistance rank 11 forces a 1/8, combined with the forced 1/2, resulting in a 1/16. In the case of status effects, this means an insta-fail.

Reply:
At 18:04 Soporific applies sleep to Para so it looks like it is succeptible to Dark sleep

https://youtu.be/beJ9ca8aZYM?si=Sq1YFiczwdajb5v7&t=1084

Reply:
I dunnow how I missed that.

In any case, now this is an issue.
This is the first documented case (that ive seen) of a mob being ACTUALLY immune to one sleep kind but not the other. Regularly, as I said before, its just a case of one of the 2 never landing, giving the regular "resists" message (not completely resists)

Now we will have to go to retail and check that those spells actually apply the same effect id or not.

For the time being, this can stay as is, however, I would change the comment to "TODO: Mob is immune to light based sleep, but not dark"

Reply:
added TODO comment 👍🏻

Reply:
Well then…that’s interesting…

Reply:
related: immunities are normally added via `mob_pools.sql` and not in script. being able to handle immunities in script is a very recent thing

the full list of values that can be used in the bit mask lives in [battleentity.h](https://github.com/LandSandBoat/server/blob/62f01f3f490833aad7c356e4c5f6918b6816d23d/src/map/entities/battleentity.h#L457) and _does_ have an entry for element sleeps but we don't want that used just yet as it was originally allowed for different reasons and we're not sure that's how we want to handle this going forward.

Reply:
Thanks for the insight @TeoTwawki ! Will keep that in mind, for this particular case I'll keep the TODO pending direction on implementation